### PR TITLE
Add e2e test cases for the operator

### PR DESCRIPTION
**Pull request by dagrayvid**
Added three tests so far:
- Check that the operator Deployment has 1 ready replica
- Check that the SRO ClusterOperator is available
- Check that the Cluster Operator is upgradeable (currently failing, not yet implemented in the operator)

Planning to add a test to verify that metrics are being published.
@zvonkok @ArangoGutierrez PTAL

### CODE REVIEW

File: config/recipes/driver-container-base/manifests/0000-driver-container-base-build.yaml [hunk 1]
```diff
@@ -107,12 +107,14 @@ data:
 
     # Installation order is important leave this as is 
     kernel_rpms=(
-      $(find /extensions -name kernel-core-${UNAME}.rpm -exec ls {} \; | tail -n1)
-      $(find /extensions -name kernel-devel-${UNAME}.rpm -exec ls {} \; | tail -n1)
-      $(find /extensions -name kernel-headers-${UNAME}.rpm -exec ls {} \; | tail -n1) 
-      $(find /extensions -name kernel-modules-${UNAME}.rpm -exec ls {} \; | tail -n1)
-      $(find /extensions -name kernel-modules-extra-${UNAME}.rpm -exec ls {} \; | tail -n1)
+      $(find /extensions -name kernel-core-${kernel_version}.rpm -exec ls {} \; | tail -n1)
```
Comment (zvonkok) [inline config/recipes/driver-container-base/manifests/0000-driver-container-base-build.yaml]: Nice catch!

File: test/e2e/basic/simple-kmod.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+			if len(skDaemonSets.Items) == 1 {
+				return true, nil
+			}
+
+			return false, nil
+		})
+
+		ginkgo.By(fmt.Sprintf("simple-kmod is ready"))
+
+		//How can we check if the module is actually loaded... or check logs of simple-kmod??
```
Comment (zvonkok) [inline test/e2e/basic/simple-kmod.go]: NTO has the same problem, it has to check if some settings are applied to the node. Ping @jmencak to see how he has done it.

File: test/framework/create_from_yaml.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+	for scanner.Scan() {
+		yamlSpec := scanner.Bytes()
+
+		obj := getObjFromYAMLSpec(yamlSpec)
+
+		err := cl.Update(context.TODO(), obj)
+		exit.OnError(err)
+
+		log.Info("Created", "Kind", obj.GetKind(), "Name", obj.GetName())
```
Comment (zvonkok) [inline test/framework/create_from_yaml.go]:
```suggestion
		log.Info("Updated", "Kind", obj.GetKind(), "Name", obj.GetName())
```

File: config/recipes/driver-container-base/manifests/0000-driver-container-base-build.yaml (hunk 1 above)
Comment (dagrayvid) [reply config/recipes/driver-container-base/manifests/0000-driver-container-base-build.yaml]: This only fixes part of the problem though, see my slack message when you get a chance

File: test/e2e/basic/simple-kmod.go (hunk 2 above)
Comment (jmencak) [reply test/e2e/basic/simple-kmod.go]: I believe I had two choices, @zvonkok .  Either use the privileged `openshift-tuned` container or just use `oc debug node`.  I've chosen the first variant as it does basically the same thing and I didn't want to start yet another container.

File: config/recipes/lustre-client/manifests/0000-buildconfig.yaml [hunk 4]
```diff
@@ -30,6 +30,8 @@ spec:
 
       ARG RELEASEVER
 
+      RUN yum list installed |  grep kernel
```
Comment (zvonkok) [inline config/recipes/lustre-client/manifests/0000-buildconfig.yaml]: What is this for?

Comment (dagrayvid) [reply config/recipes/lustre-client/manifests/0000-buildconfig.yaml]: Oops. This is from some testing I did. Good catch

**Pull request merged**